### PR TITLE
Use $scope observables for dependency-based data flow

### PR DESCRIPTION
**Pull request by uniphil**

### CODE REVIEW

File: app/scripts/services/opendataapi.coffee [hunk 1]
```diff
@@ -23,6 +23,29 @@ angular.module 'edudashAppSrv'
           'improvement': 'bba2cbbb-97fb-48b1-aa51-8db69279fbc5'
         secondary: '743e5062-54ae-4c96-a826-16151b6f636b'
 
+      converters =
```
Comment (esoliani) [inline app/scripts/services/opendataapi.coffee:26]: Nice way to parse ckan data

File: app/views/ratepasschart.html [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-            <div class="pass-rate" ng-class="{'girl-pass-grey': (selectedSchool.PASS_RATE/10 + $index) % 2 != 0,
-                                              'boy-pass-grey': (selectedSchool.PASS_RATE/10 + $index) % 2 == 0}"
-                 ng-repeat="i in getTimes(10-(selectedSchool.PASS_RATE/10)) track by $index"></div>
-        </div>
+  <p class="widgettitle">{{selectedYear}} {{'chart.title.passrate-2014'|translate}}</p>
+  <div class="row" ng-if="selectedSchool.PASS_RATE != '0'">
+    <div ng-class="{'col-xs-3': selectedSchool.PASS_RATE == '100', 'col-xs-2': selectedSchool.PASS_RATE != '100'}">
+      <p class="percentNumber widgetnumber">{{ selectedSchool.PASS_RATE| number:0}}%</p>
     </div>
-    <div class="row" ng-if="selectedSchool.PASS_RATE == '0'">
```
Comment (esoliani) [inline app/views/ratepasschart.html]: why do you remove the NA text? the condition should changed

File: app/views/ratepasschart.html [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+          ng-class="getClass($index, selectedSchool.PASS_RATE, max, min)"
+          ng-repeat="i in getTimes(selectedSchool.PASS_RATE/10) track by $index"></div>
+      <div
+          class="pass-rate"
+          ng-class="{'girl-pass-grey': (selectedSchool.PASS_RATE/10 + $index) % 2 != 0,
+                     'boy-pass-grey': (selectedSchool.PASS_RATE/10 + $index) % 2 == 0}"
+          ng-repeat="i in getTimes(10-(selectedSchool.PASS_RATE/10)) track by $index"></div>
     </div>
+  </div>
+  <div class="row" ng-if="selectedSchool.PASS_RATE == '0'">
```
Comment (esoliani) [inline app/views/ratepasschart.html]: does this condition works? since the pass_rate is numeric now

File: app/views/schoollist.html [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+    <tr ng-repeat="school in schools">
+      <td class="rank">{{ $index + 1 }}.</td>
+      <td class="name"
+          ng-click="click(school.CODE)"
+          ng-mouseenter="hover(school.CODE)"
+          ng-mouseleave="unHover()">
+        <strong>{{ school.NAME }}</strong>
+      </td>
+      <td ng-class="{
+        'passrategreen': school[property]>=(max+0),
```
Comment (esoliani) [inline app/views/schoollist.html]: I think we could remove this conversion

File: app/views/ratepasschart.html (hunk 3 above)
Comment (uniphil) [inline app/views/ratepasschart.html]: haha it does, but it's terrible and thanks for catching it. `==` will do implicit conversions in JS. I'll swap it out for `===` and make it a literal zero

**Comment by ddimunzio** [comment]
let's merge and fix issues afterwards

**Pull request merged**